server/re/src/irods_re_serialization.cpp

```diff
@@ -35,6 +40,21 @@ namespace irods::re_serialization
         }
     }
 
+    static error serialize_keyValPair(boost::any _p, serialized_parameter_t& _out)
+    {
+        try {
+            const keyValPair_t& kvp = boost::any_cast<keyValPair_t>(_p);
+            serialize_keyValPair(kvp, _out);
```

Comment:
c++ noob here... how could this work?
if it succeeded in casting to a keyValPair... then we call this same function again on the result?

Reply:
I'll give you a hint.

Look at the function signature for the other `keyValPair_t` function. It's above this one.

Reply:
it's a ref!

as you were.

Reply:
The compiler will pick the function overload which best matches the types involved.